runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxOverlay.java

```diff
@@ -104,9 +115,59 @@ public Dimension render(Graphics2D graphics)
 				}
 			}
 
-			x += BOXSIZE + SEPARATOR;
+			// Determine which axis to reset/increase
+			if (vertical)
+			{
+				// Reset y if newbox reaches height limit
+				if (y + TOTAL_BOXSIZE < height)
+				{
+					y += TOTAL_BOXSIZE;
+				}
+				else
+				{
+					y = 0;
+					x += TOTAL_BOXSIZE;
+				}
+			}
+			else
+			{
+				// Reset x if newbox reaches width limit
+				if (x + TOTAL_BOXSIZE < width)
+				{
+					x += TOTAL_BOXSIZE;
+				}
+				else
+				{
+					x = 0;
+					y += TOTAL_BOXSIZE;
+				}
+			}
+		}
+
+		return new Dimension(width, height);
+	}
+
+	private int getFirstSize(int infoBoxSize, int maxRow)
```

Comment:
FirstSize and SecondSize are a bit confusing, maybe getHorizontalWrap and getVerticalWrap